tracpro/contacts/models.py

```diff
@@ -97,10 +100,12 @@ def kwargs_from_temba(cls, org, temba_contact):
 
         org_region_uuids = [r.uuid for r in Region.get_all(org)]
         region_uuids = intersection(org_region_uuids, temba_contact.groups)
-        region = Region.objects.get(org=org, uuid=region_uuids[0]) if region_uuids else None
-
-        if not region:  # pragma: no cover
-            raise ValueError("No region with UUID in %s" % ", ".join(temba_contact.groups))
+        if len(region_uuids) > 0:
+            region = Region.objects.filter(org=org, uuid__in=region_uuids).first() if region_uuids else None
+            if not region:  # pragma: no cover
```

Comment:
Am not sure I understand what we're trying to do here. I think the original exception was occurring because the contact was not in any of the configured region groups, and region is a required field for a contact. `region_uuids` is the intersection of the set of all region group UUIDs, and of the incoming contact's group UUIDs. If `region_uuids` is not empty, then `region` will never be falsey. I may be missing something but I think all this does is change the exception.